src/middleware/language.js

```diff
@@ -4,30 +4,39 @@ const parser = require('accept-language-parser');
 const _ = require('lodash');
 const { languageCodeExists } = require('country-language');
 
+const DEFAULT_LANGUAGE = 'en';
+
 module.exports = setup;
 
 /**
- * Find matching language by partial match.
+ * Intersect the list of user accept languages over the list of authorized languages
+ * and return the first matching language
  *
- * To match, the language must have the same country code.
+ * iterate over the ordered list of user accepted languages and:
+ *   1/ find exact match (country + region)
+ *   2/ find partial match (country code)
  *
- * @param   {Array} languages Array of known languages.
- * @param   {Array} parsedLanguages Array of parsed languages from Accept-Language.
+ * @param   {Array} authorizedLanguages Array of authorized/known languages.
+ * @param   {Array} userAcceptLanguages Array of parsed languages from Accept-Language.
```

Comment:
I think `userAcceptLanguages ` should be parsed/ordered inside this function. Because inside the function we assume it is ordered but in the future, if it is done outside the function it is not a guarantee.